### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by jsun969**
Run `./test_runner` in build folder to test

### CODE REVIEW

File: src/Game.cpp [hunk 1]
```diff
@@ -15,33 +15,49 @@ Game::Game() : enemies(), objects(), playingUI(), mainMenu(), pauseMenu(), deadM
 	addUIObject(new FPSCounter(this));
 	player = new Player(this);
 
+	// Load font
+#ifdef BUILD_TEST_RUNNER
```
Comment (xxori) [inline src/Game.cpp]: can this be ifndef? Why is there an empty ifdef block

File: src/test/ButtonTest.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#include "Color.hpp"
+#include <iostream>
+
+ButtonTest::ButtonTest(Game* game) {
+	button = new Button(game, nullptr, raylib::Vector2 { 0, 0 }, RAYWHITE, RAYWHITE, "Test Button", 6, 6, RAYWHITE);
+}
+
+ButtonTest::~ButtonTest() {
+	delete button;
+	button = nullptr;
```
Comment (xxori) [inline src/test/ButtonTest.cpp]: dont think theres any point setting it to nullptr in destructor

File: src/test/EnemyTest.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	enemy = nullptr;
+	delete enemyWithCooldown;
+	enemyWithCooldown = nullptr;
+}
+
+void EnemyTest::testHealth() {
+	std::cout << "Testing: EnemyTest::testHealth\n";
+	enemy->getHealth();
+	bool pass { enemy->getHealth() == 6 };
+	if (pass) {
```
Comment (xxori) [inline src/test/EnemyTest.cpp:25]: we dont really need to set this to a new variable, can just be inline

**Review by xxori — COMMENTED** [review]
Good work! Very comprehensive

File: src/Game.cpp (hunk 1 above)
Comment (jsun969) [reply src/Game.cpp]: yea I'll change it

File: src/test/ButtonTest.cpp (hunk 2 above)
Comment (jsun969) [reply src/test/ButtonTest.cpp]: just a best practice ig

File: src/test/EnemyTest.cpp (hunk 3 above)
Comment (jsun969) [reply src/test/EnemyTest.cpp:25]: ehh, that look too long for some of the conditions

**Review by xxori — APPROVED** [review]

**Pull request merged**